plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePartitionsTable.java

```diff
@@ -0,0 +1,470 @@
+
+            public Builder(List<DeltaLakeColumnHandle> columns, TypeManager typeManager)
+            {
+                this.columns = ImmutableList.copyOf(requireNonNull(columns, "columns is null"));
+                this.typeManager = requireNonNull(typeManager, "typeManager is null");
+            }
+
+            public void acceptAddFileEntry(AddFileEntry addFileEntry)
+            {
+                if (addFileEntry.getDeletionVector().isPresent()) {
+                    return;
+                }
+
+                fileCount++;
+                size += addFileEntry.getSize();
+
+                if (ignoreDataColumn) {
+                    return;
+                }
+
+                addFileEntry.getStats().ifPresentOrElse(stats -> {
+                    for (DeltaLakeColumnHandle column : columns) {
+                        updateMinMaxStats(
+                                column.getBaseColumnName(),
+                                column.getType(),
+                                stats.getMinColumnValue(column).orElse(null),
+                                stats.getMaxColumnValue(column).orElse(null),
+                                Optional.of(0L),
+                                stats.getNumRecords().orElse(0L));
+                        updateNullCountStats(column.getBaseColumnName(), Optional.of(stats.getNullCount(column.getBaseColumnName()).orElse(0L)));
+                    }
+                }, () -> {
+                    columnStatistics.clear();
+                    nullCounts.clear();
+                    ignoreDataColumn = true;
+                });
+            }
+
+            public DeltaLakePartitionStatistics build()
+            {
+                ImmutableMap.Builder<String, Object> minValues = ImmutableMap.builder();
+                ImmutableMap.Builder<String, Object> maxValues = ImmutableMap.builder();
+
+                columnStatistics.forEach((key, statistics) -> {
+                    statistics.getMin().ifPresent(min -> minValues.put(key, min));
+                    statistics.getMax().ifPresent(max -> maxValues.put(key, max));
+                });
+
+                Map<String, Long> nullCounts = this.nullCounts.entrySet().stream()
+                        .filter(entry -> entry.getValue().isPresent())
+                        .collect(toImmutableMap(Map.Entry::getKey, entry -> entry.getValue().orElseThrow()));
+
+                return new DeltaLakePartitionStatistics(fileCount, size, minValues.buildOrThrow(), maxValues.buildOrThrow(), nullCounts);
+            }
+
+            private void updateNullCountStats(String key, Optional<Long> nullCount)
+            {
+                // If one file is missing nullCounts for a column, invalidate the estimate
+                nullCounts.merge(key, nullCount, (existingCount, newCount) ->
+                        existingCount.isPresent() && newCount.isPresent() ? Optional.of(existingCount.get() + newCount.get()) : Optional.empty());
```

Comment:
The comment is obvious when reading the code. It would be nice to leave a reason why "invalidate the estimate" in such situation.